### PR TITLE
added the new and create in yachts controller

### PR DESCRIPTION
**Pull request by Cami55**

### CODE REVIEW

File: app/controllers/yachts_controller.rb [hunk 1]
```diff
@@ -6,8 +6,15 @@ def show
   end
 
   def new
+    @yacht = Yacht.new
   end
 
   def create
+    @yacht = Yacht.new(yacht_params)
+    if @yachts.save
+      redirect_to @yacht
```
Comment (galasoa) [inline app/controllers/yachts_controller.rb:15]: I think it's supposed to be `redirect_to yacht_path(@yacht)` here

File: app/controllers/yachts_controller.rb [hunk 2]
```diff
@@ -10,11 +10,9 @@ def new
...
+     @yacht = Yacht.new(yacht_params)
     if @yachts.save
-      redirect_to @yacht
-    else
-      render :new
-    end
+      redirect_to yacht_path(@yacht)
   end
 end
+end
```
Comment (galasoa) [inline app/controllers/yachts_controller.rb:18]: extra end

Comment (barangerbenjamin) [reply app/controllers/yachts_controller.rb:18]: No extra `end` here, but careful at the indentation 👍

File: app/controllers/yachts_controller.rb [hunk 3]
```diff
@@ -6,8 +6,13 @@ def show
   end
 
   def new
+    @yacht = Yacht.new
   end
 
   def create
+     @yacht = Yacht.new(yacht_params)
+    if @yachts.save
```
Comment (barangerbenjamin) [inline app/controllers/yachts_controller.rb:14]: Should be `yacht` same as line 13

Comment (barangerbenjamin) [reply app/controllers/yachts_controller.rb:14]: Remember, before the `.save`, what does a yacht belongs_to ? A user
So we need an extra line
`@yacht.user = curent_user`

Meaning only a logged_in user should be able to access the form.
So we need to protect this controller action behind an `authenticate_user` 👍

**Pull request merged**